forest/cli/src/cli/snapshot_cmd.rs

```diff
@@ -1,147 +1,85 @@
 // Copyright 2019-2023 ChainSafe Systems
 // SPDX-License-Identifier: Apache-2.0, MIT
 
-use std::{fs, path::PathBuf, sync::Arc, time::Duration};
+use std::{path::PathBuf, sync::Arc, time::Duration};
 
-use anyhow::bail;
-use chrono::{Datelike, Utc};
+use anyhow::{bail, Context as _};
+use chrono::Utc;
 use clap::Subcommand;
 use dialoguer::{theme::ColorfulTheme, Confirm};
 use forest_blocks::{tipset_keys_json::TipsetKeysJson, Tipset, TipsetKeys};
 use forest_chain::ChainStore;
-use forest_cli_shared::cli::{
-    default_snapshot_dir, is_car_or_zst_or_tmp, snapshot_fetch, SnapshotServer, SnapshotStore,
-};
+use forest_cli_shared::snapshot::SnapshotMetadata;
 use forest_db::db_engine::{db_root, open_proxy_db};
 use forest_genesis::{forest_load_car, read_genesis_header};
 use forest_ipld::{recurse_links_hash, CidHashSet};
 use forest_networks::NetworkChain;
 use forest_rpc_api::{chain_api::ChainExportParams, progress_api::GetProgressType};
 use forest_rpc_client::{chain_ops::*, progress_ops::get_progress};
 use forest_utils::{
-    io::{parser::parse_duration, ProgressBar},
+    io::{parser::parse_duration, progress_bar::downloading_style, ProgressBar},
     net::get_fetch_progress_from_file,
-    retry,
+    retry, RetryArgs,
 };
 use fvm_shared::clock::ChainEpoch;
-use log::info;
-use strfmt::strfmt;
+use itertools::Itertools as _;
 use tempfile::TempDir;
-use tokio::time::sleep;
+use tracing::error;
 
 use super::*;
 use crate::cli::{cli_error_and_die, handle_rpc_err};
 
-pub(crate) const OUTPUT_PATH_DEFAULT_FORMAT: &str =
-    "forest_snapshot_{chain}_{year}-{month}-{day}_height_{height}.car";
-
-pub(crate) const OUTPUT_PATH_DEFAULT_COMPRESSED_FORMAT: &str =
-    "forest_snapshot_{chain}_{year}-{month}-{day}_height_{height}.car.zst";
-
 #[derive(Debug, Subcommand)]
 pub enum SnapshotCommands {
     /// Export a snapshot of the chain to `<output_path>`
     Export {
         /// Snapshot output path. Default to
-        /// `forest_snapshot_{chain}_{year}-{month}-{day}_height_{height}.car(.
-        /// zst)` Date is in ISO 8601 date format.
-        /// Arguments:
-        ///  - chain - chain name e.g. `mainnet`
-        ///  - year
-        ///  - month
-        ///  - day
-        ///  - height - the epoch
+        /// `forest_snapshot_{chain}_{year}-{month}-{day}_height_{epoch}.car.
+        /// zst`.
         #[arg(short, default_value = ".", verbatim_doc_comment)]
         output_path: PathBuf,
-        /// Export in zstd compressed format
-        #[arg(long)]
-        compressed: bool,
-        /// Skip creating the checksum file. Only valid when `--compressed` is
-        /// not supplied.
+        /// Skip creating the checksum file.
         #[arg(long)]
         skip_checksum: bool,
```

Comment:
In general, given that this compression has in-built poor man's checksum, we could eliminate this altogether. Let's address it in another issue, though.

Reply:
#2933